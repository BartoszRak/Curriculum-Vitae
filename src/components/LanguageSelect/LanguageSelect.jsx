import React from 'react'
import PropTypes from 'prop-types'
import { withStyles, Select, MenuItem } from '@material-ui/core'

import { LanguageContext } from '~services/Internacionalization'

import style from './LanguageSelect.style'

export function LanguageSelect({ classes }) {
  return (
    <LanguageContext.Consumer>
      {({ language, setLanguage }) => (
        <div className={classes.root}>
          <Select
            classes={{
              root: classes.select,
            }}
            name="Language"
            onChange={evt => {
              setLanguage(evt.target.value)
            }}
            value={language}
            variant="standard"
          >
            <MenuItem value="en">English</MenuItem>
            <MenuItem value="pl">Polish</MenuItem>
          </Select>
        </div>
      )}
    </LanguageContext.Consumer>
  )
}

LanguageSelect.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(LanguageSelect)
