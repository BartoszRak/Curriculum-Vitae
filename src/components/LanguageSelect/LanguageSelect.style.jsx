import { makeStyles } from '@material-ui/core'

export default makeStyles(({ palette, spacing }) => ({
  menuPaper: {
    '& li': {
      fontSize: '0.9rem',
      padding: spacing(0.5, 3),
    },
  },
  root: {},
  select: {
    '&:after': {
      display: 'none',
    },
    '&:before': {
      display: 'none',
    },
    background: 'none',
  },
  selectIcon: {
    color: palette.primary.main,
    marginRight: spacing(1),
  },
  selectMenu: {
    '&:focus': {
      background: palette.secondary.main,
      borderRadius: 100,
    },
  },
  selectRoot: {
    background: palette.secondary.main,
    borderRadius: 100,
    color: palette.primary.main,
    fontSize: '0.9rem',
    padding: spacing(0.5, 5, 0.5, 2),
  },
}))
