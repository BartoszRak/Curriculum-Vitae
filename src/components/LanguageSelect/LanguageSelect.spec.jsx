import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { LanguageSelect } from './LanguageSelect'

jest.unmock('@material-ui/core')
jest.mock('~services/Internacionalization')
jest.mock('react-intl')

describe('component: LanguageSelect', () => {
  afterEach(cleanup)

  describe('rendering', () => {
    test('match snapshot', () => {
      const wrapper = render(<LanguageSelect />)

      expect(wrapper).toMatchSnapshot()
    })

    test('render select', () => {
      const { getByTestId } = render(<LanguageSelect />)
      const el = getByTestId('language-select')

      expect(el).toBeVisible()
    })
  })
})
