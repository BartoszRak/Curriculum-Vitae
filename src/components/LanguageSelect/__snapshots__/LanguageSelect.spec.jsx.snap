// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: LanguageSelect rendering match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-2"
      >
        <div
          aria-label="Choose language"
          class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-3"
          data-testid="language-select"
        >
          <div
            class="MuiSelect-root makeStyles-selectRoot-6"
          >
            <div
              aria-haspopup="true"
              aria-pressed="false"
              class="MuiSelect-select MuiSelect-selectMenu makeStyles-selectMenu-5 MuiInputBase-input MuiInput-input"
              id="select-Language"
              role="button"
              tabindex="0"
            />
            <input
              name="Language"
              type="hidden"
              value="en"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon makeStyles-selectIcon-4"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-2"
    >
      <div
        aria-label="Choose language"
        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-select-3"
        data-testid="language-select"
      >
        <div
          class="MuiSelect-root makeStyles-selectRoot-6"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            class="MuiSelect-select MuiSelect-selectMenu makeStyles-selectMenu-5 MuiInputBase-input MuiInput-input"
            id="select-Language"
            role="button"
            tabindex="0"
          />
          <input
            name="Language"
            type="hidden"
            value="en"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon makeStyles-selectIcon-4"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
